staging/src/k8s.io/apimachinery/pkg/conversion/converter.go

```diff
@@ -485,6 +487,12 @@ func (c *Converter) DefaultConvert(src, dest interface{}, flags FieldMatchingFla
 type conversionFunc func(sv, dv reflect.Value, scope *scope) error
 
 func (c *Converter) doConversion(src, dest interface{}, flags FieldMatchingFlags, meta *Meta, f conversionFunc) error {
+	_, srcUnstructured := src.(runtime.Unstructured)
```

Comment:
move this to just above the `// TODO: consider everything past this point deprecated` line... that way we don't slow down the existing conversion invocations that use generated functions